src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -449,6 +459,16 @@ private Task WriteFormattedInt32Async(int value, CancellationToken cancellationT
                     responseStream = EmptyReadStream.Instance;
                     ReturnConnectionToPool();
                 }
+                else if (request.Method == s_httpConnectMethod && (int)response.StatusCode == 200)
```

Comment:
I think we should check for request.Method == s_httpConnectMethod once up front and store it in a bool.  The string comparison isn't that expensive, but it's better to do it just once.

Reply:
ok. Would that change if we can simply compare instances? s_httpConnectMethod may not be private string at some point, right?

Reply:
Yeah, s_httpConnectMethod could be made public -- maybe should be made public.

That said, this doesn't prevent a user from creating their own instance of HttpMethod with "CONNECT" as the string value.

I don't really see any reason not to do the check once and store the bool for later usage.